src/shared/components/submit-button.tsx

```diff
@@ -20,6 +20,7 @@ export const SubmitButton: React.FC<IProps> = ({ isAnswered }) => {
 
   const handleSubmit = () => {
     setInteractiveState?.(prevState => ({...prevState, submitted: true }));
```

Comment:
So in this case, with an open response will there be an 'answer saved' event and a 'answer submitted' event since the focus would have left the the text box as the user clicks the submit button?
And in that case which one happens first?  
I'm wondering about the order because it seems possible the `submitted: true` appears to be unset in the second event if the order is a certain way.

Reply:
> I'm wondering about the order because it seems possible the submitted: true appears to be unset in the second event if the order is a certain way.

We don't log interactive state anyway, so submitted value isn't visible in the logs.
If you type in the box, move your mouse directly to Submit button, and click, the order of events is:
- "answer submitted", answer_text: "test"
- "focus out"
- "answer saved", answer_text: "test"

Generally saving on blur / focus out isn't perfect. I think I could check if there's submit=true in interactive state and don't trigger "answer saved" even then.

Reply:
Nah, it is probably better to keep the answer saved logic simple so there is less chances for failure. 
Yeah the focus tracking isn't great, but I don't know how else to generically track user interaction.